ksqldb-engine/src/main/java/io/confluent/ksql/services/DefaultConnectClient.java

```diff
@@ -344,6 +343,25 @@ private String resolveUri(final String relativePath) {
     }
   }
 
+  private static Header[] buildHeaders(
+      final Optional<String> authHeader,
+      final Map<String, String> additionalRequestHeaders
+  ) {
+    final List<Header> headers = new ArrayList<>();
+
+    authHeader.ifPresent(header -> headers.add(
+        new BasicHeader(HttpHeaders.AUTHORIZATION.toString(), authHeader.get())));
+
+    if (!additionalRequestHeaders.isEmpty()) {
+      final List<Header> additionalHeaders = additionalRequestHeaders.entrySet().stream()
+          .map(e -> new BasicHeader(e.getKey(), e.getValue()))
+          .collect(Collectors.toList());
+      headers.addAll(additionalHeaders);
+    }
```

Comment:
This is related to the comment of the headers precedence. If I set an authHeader in additionalHeaders, then this code is going to override it. Should we keep the default ksql header or allow the user to override it?

Reply:
Hm good question. I could see the argument going either way. Currently, I think it's nice for users to be able to override the auth header if they want. This could be used as a workaround for supporting custom forms of authN other than basic auth, since that's currently the only form we support today. However, I could see this override behavior being confusing in the future, if more ksql headers are introduced.

I'm leaning towards leaving the behavior for now, unless you feel otherwise.